kube/src/runtime/informer.rs

```diff
@@ -58,94 +64,79 @@ where
 
     /// Start a single watch stream
     ///
-    /// Opens a long polling GET and returns the complete WatchEvents as a Stream.
+    /// Opens a long polling GET and returns a stream of WatchEvents.
     /// You should always poll. When this call ends, call it again.
     /// Do not call it from more than one context.
     ///
-    /// This function will handle error handling up to a point:
-    /// - if we go out of history (410 Gone), we reset to latest
-    /// - if we failed an initial poll, we will retry
     /// All real errors are bubbled up, as are WachEvent::Error instances.
-    /// In the retry/reset cases we wait 10s between each attempt.
+    /// If we are desynced we force a 10s wait 10s before starting the poll.
     ///
     /// If you need to track the `resourceVersion` you can use `Informer::version()`.
     pub async fn poll(&self) -> Result<impl Stream<Item = Result<WatchEvent<K>>>> {
-        trace!("Watching {}", self.resource.kind);
+        trace!("Watching {}", self.api.resource.kind);
 
         // First check if we need to backoff or reset our resourceVersion from last time
         {
-            let mut needs_retry = self.needs_retry.lock().await;
             let mut needs_resync = self.needs_resync.lock().await;
-            if *needs_resync || *needs_retry {
+            if *needs_resync {
                 // Try again in a bit
                 let dur = Duration::from_secs(10);
-                delay_for(dur).await;
+                tokio::time::delay_for(dur).await;
                 // If we are outside history, start over from latest
                 if *needs_resync {
                     self.reset().await;
                 }
                 *needs_resync = false;
-                *needs_retry = false;
             }
         }
 
-        // Create our watch request
-        let resource_version = self.version.lock().await.clone();
-        let req = self.resource.watch(&self.params, &resource_version)?;
-
         // Clone Arcs for stream handling
         let version = self.version.clone();
         let needs_resync = self.needs_resync.clone();
 
-        // Attempt to fetch our stream
-        let stream = self.client.request_events::<K>(req).await;
+        // Start watching from our previous watch point
+        let resource_version = self.version.lock().await.clone();
+        let stream = self.api.watch(&self.params, &resource_version).await?;
```

Comment:
this question mark is what causes the initial Ok match to drop away. (We are no longer doing anything in the error case on behalf of the Informer anyway).